### ISSUE TITLE
Create `MATHESAR_URI` custom type and handle it in the backend.

### ISSUE THREAD

**Issue by kgodey** [centerofci/mathesar#412]

**Comment by mathemancer** [comment]
If we can spare the time for implementation, I suggest we implement our own.  This would avoid the need for installing a 3rd-party extension which isn't supported everywhere (e.g., on Amazon's RDS instances).

**Comment by kgodey** [comment]
`pguri` is not on [AWS's supported extension list](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.FeatureSupport.Extensions.13x).

We might still want to use it to speed up implementation and consider writing our own at a later date. It depends on how much work it is to write our own. We could also try using/forking `pguri` as a starting point if we're writing our own.

**Comment by kgodey** [comment]
@mathemancer makes sense. I do want to make it easy to extend to other URI parts in the future (e.g. `pguri` supports port, query parameters, fragments, etc. (see [docs](https://github.com/petere/pguri#using)).

**Comment by mathemancer** [comment]
> @mathemancer makes sense. I do want to make it easy to extend to other URI parts in the future (e.g. `pguri` supports port, query parameters, fragments, etc. (see [docs](https://github.com/petere/pguri#using)).

I do think it makes sense to try to use that extension (and other URI-mangling libraries) as inspiration at least.  I just wish there was a good way to install extensions without access to the underlying server.
